testcrate/tests/os_version_check.rs

```diff
@@ -0,0 +1,153 @@
+#![cfg(target_vendor = "apple")]
+use std::process::Command;
+
+use compiler_builtins::os_version_check::__isOSVersionAtLeast;
+
+#[test]
+fn test_general_available() {
+    // Lowest version always available.
+    assert_eq!(__isOSVersionAtLeast(0, 0, 0), 1);
+    // This high version never available.
+    assert_eq!(__isOSVersionAtLeast(9999, 99, 99), 0);
+}
+
+#[test]
+#[cfg_attr(
+    not(target_os = "macos"),
+    ignore = "`sw_vers` is only available on macOS"
+)]
+fn compare_against_sw_vers() {
+    let sw_vers = Command::new("sw_vers")
+        .arg("-productVersion")
+        .output()
+        .unwrap()
+        .stdout;
+    let sw_vers = String::from_utf8(sw_vers).unwrap();
+    let mut sw_vers = sw_vers.trim().split('.');
+
+    let major: u32 = sw_vers.next().unwrap().parse().unwrap();
+    let minor: u32 = sw_vers.next().unwrap_or("0").parse().unwrap();
+    let subminor: u32 = sw_vers.next().unwrap_or("0").parse().unwrap();
+    assert_eq!(sw_vers.count(), 0);
+
+    // Current version is available
+    assert_eq!(__isOSVersionAtLeast(major, minor, subminor), 1);
+
+    // One lower is available
+    assert_eq!(
+        __isOSVersionAtLeast(major, minor, subminor.saturating_sub(1)),
+        1
+    );
+    assert_eq!(
+        __isOSVersionAtLeast(major, minor.saturating_sub(1), subminor),
+        1
+    );
+    assert_eq!(
+        __isOSVersionAtLeast(major.saturating_sub(1), minor, subminor),
+        1
+    );
+
+    // One higher isn't available
+    assert_eq!(__isOSVersionAtLeast(major, minor, subminor + 1), 0);
+    assert_eq!(__isOSVersionAtLeast(major, minor + 1, subminor), 0);
+    assert_eq!(__isOSVersionAtLeast(major + 1, minor, subminor), 0);
+}
+
+// Test internals
+
+#[path = "../../src/os_version_check/darwin_impl.rs"]
+#[allow(dead_code)]
+mod darwin_impl;
```

Comment:
A bit unsure about this, should I make it a unit-test inside `src/os_version_check/darwin_impl.rs` instead of an integration test?

Reply:
Unit tests unfortunately don't really work in `compiler-builtins`. It is fine to just make everything public so you don't need to do the module path trick.